File: PlaybackLab/TVOSSampleApp/TVOSSampleApp/Players/ChildPlayerViewController.m
//
//  ChildPlayerViewController.m
//  TVOSSampleApp
//
//  Copyright © 2016 Ooyala. All rights reserved.
//

#import "ChildPlayerViewController.h"
#import "PlayerSelectionOption.h"
#import <OoyalaTVSkinSDK/OOOoyalaTVPlayerViewController.h>
#import <OoyalaSDK/OOOoyalaPlayer.h>
#import <OoyalaSDK/OOPlayerDomain.h>

@interface ChildPlayerViewController ()

@property (nonatomic, weak) IBOutlet UIView *playerView;

@property (nonatomic) NSString *pcode;
@property (nonatomic) NSString *playerDomain;

@end

@implementation ChildPlayerViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  self.pcode = self.option.pcode;
  self.playerDomain = self.option.domain;
  
  // Create Ooyala ViewController
  OOOoyalaPlayer *player = [[OOOoyalaPlayer alloc] initWithPcode:self.pcode
                                                          domain:[[OOPlayerDomain alloc] initWithString:self.playerDomain]];
  self.ooyalaPlayerViewController = [[OOOoyalaTVPlayerViewController alloc] initWithPlayer:player];
  self.ooyalaPlayerViewController.playbackControlsEnabled = NO;
  
  [NSNotificationCenter.defaultCenter addObserver:self
                                         selector:@selector(notificationHandler:)
                                             name:nil
                                           object:self.ooyalaPlayerViewController.player];
  
  // Attach it to current view
  [self addChildViewController:self.ooyalaPlayerViewController];
  self.ooyalaPlayerViewController.view.frame = self.playerView.bounds;
  [self.playerView addSubview:self.ooyalaPlayerViewController.view];
  [self.ooyalaPlayerViewController didMoveToParentViewController:self];
  
  // Load the video
  [self.ooyalaPlayerViewController.player setEmbedCode:self.option.embedCode];
  [self.ooyalaPlayerViewController.player play];
}

@end

File: PlaybackLab/TVOSSwiftSampleApp/TVOSSwiftSampleApp/Players/ChildPlayerViewController.swift
//
//  ChildPlayerViewController.swift
//  TVOSSwiftSampleApp
//
//  Copyright © 2018 Ooyala. All rights reserved.
//
// This example shows shows a use case in which you want to show the player next to
// other UI in the screen, so the player will not be in fullscreen mode taking all
// your screen space, but it will be sharing the UI with other elements which could
// be actionable too, like buttons.
//
// This is not a verified use case by Ooyala and cause extrange behavior for the player,
// but we provide the example in case you want to use it.

import Foundation

class ChildPlayerViewController: AbstractPlayerViewController {
  
  @IBOutlet var playerView: UIView!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
     // Create Ooyala ViewController
    let player = OOOoyalaPlayer(pcode: option.pcode,
                                domain: OOPlayerDomain(string: option.domain))
    ooyalaPlayerViewController = OOOoyalaTVPlayerViewController(player: player)
    ooyalaPlayerViewController.playbackControlsEnabled = false
    
    NotificationCenter.default.addObserver(self,
                                           selector: #selector(notificationHandler(_:)),
                                           name: nil,
                                           object: player)
    
    // Attach it to current view
    addChild(ooyalaPlayerViewController)
    ooyalaPlayerViewController.view.frame = playerView.bounds
    playerView.addSubview(ooyalaPlayerViewController.view)
    ooyalaPlayerViewController.didMove(toParent: self)
    
    // Load the video
    ooyalaPlayerViewController.player.setEmbedCode(option.embedCode, shouldAutoPlay: true, withCallback: nil)
  }
  
}
